Mathlib/Analysis/Normed/Group/Ultrametric.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Analysis.Normed.Group.Uniform
+import Mathlib.Topology.Algebra.Nonarchimedean.Basic
+import Mathlib.Topology.MetricSpace.Ultra.Basic
+
+/-!
+# Ultrametric normed groups
+
+We give formulations of the ultrametric triangle inequality for ultrametric spaces where the metric
+is induced by a norm; and prove that such groups are nonarchimedean groups (which is *not* true by
+definition in Mathlib – the definition of `NonarchimedeanGroup` does not refer to metrics or norms).
+-/
+
+namespace IsUltrametricDist
+
+/-- Triangle inequality for ulrametric (semi) norms on groups. -/
+@[to_additive "Triangle inequality for ulrametric (semi) norms on additive groups."]
+lemma norm_mul_le_max
```

Comment:
Can you add the `nnnorm` version too?

Reply:
I've multiplicativized #14768 so that this and related lemmas are there